pymc3/plots/posteriorplot.py

```diff
@@ -1,71 +0,0 @@
-#   Copyright 2020 The PyMC Developers
-#
-#   Licensed under the Apache License, Version 2.0 (the "License");
-#   you may not use this file except in compliance with the License.
-#   You may obtain a copy of the License at
-#
-#       http://www.apache.org/licenses/LICENSE-2.0
-#
-#   Unless required by applicable law or agreed to in writing, software
-#   distributed under the License is distributed on an "AS IS" BASIS,
-#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#   See the License for the specific language governing permissions and
-#   limitations under the License.
-
-from __future__ import annotations
-
-from typing import TYPE_CHECKING, Any, Callable, Optional, Union
-
-import matplotlib.pyplot as plt
-import numpy as np
-
-from pymc3.backends.base import MultiTrace
-
-if TYPE_CHECKING:
-    from arviz.data.inference_data import InferenceData
-
-
-def plot_posterior_predictive_glm(
```

Comment:
this exact function isn't (yet) available in arviz and so perhaps shouldn't be removed just yet (this is already mentioned in the issue, just repeating it here for verbosity in case it's overlooked)

Reply:
Yes, looks like the discussion in https://github.com/pymc-devs/pymc3/issues/4371 settled on keeping just `plot_posterior_predictive_glm` around, but let's add a `DeprecationWarning`.

Reply:
Sounds great, I'll add that with the following:
```python
    warnings.warn(
        "The `plot_posterior_predictive_glm` function will migrate to Arviz in a future release. "
        "\nKeep up to date with `ArviZ <https://arviz-devs.github.io/arviz/>`_ for future updates.",
        DeprecationWarning,
    )
```

Reply:
Sounds good!